src/tests/ftest/util/test_utils_pool.py

```diff
@@ -255,39 +222,25 @@ def create(self):
             if value is not None:
                 kwargs[key] = value
 
-        if self.control_method.value == self.USE_API:
-            raise CommandFailure(
-                "Error: control method {} not supported for create()".format(
-                    self.control_method.value))
-
-        if self.control_method.value == self.USE_DMG and self.dmg:
-            # Create a pool with the dmg command and store its CmdResult
-            self._log_method("dmg.pool_create", kwargs)
-            data = self.dmg.pool_create(**kwargs)
-
-            if self.dmg.result.exit_status == 0:
-                # Convert the string of service replicas from the dmg command
-                # output into an ctype array for the DaosPool object using the
-                # same technique used in DaosPool.create().
-                service_replicas = [
-                    int(value) for value in data["svc"].split(",")]
-                rank_t = ctypes.c_uint * len(service_replicas)
-                rank = rank_t(*list([svc for svc in service_replicas]))
-                rl_ranks = ctypes.POINTER(ctypes.c_uint)(rank)
-                self.pool.svc = daos_cref.RankList(
-                    rl_ranks, len(service_replicas))
-
-                # Set UUID and attached to the DaosPool object
-                self.uuid = data["uuid"]
-                self.pool.attached = 1
-
-        elif self.control_method.value == self.USE_DMG:
-            self.log.error("Error: Undefined dmg command")
-
-        else:
-            self.log.error(
-                "Error: Undefined control_method: %s",
-                self.control_method.value)
+        # Create a pool with the dmg command and store its CmdResult
+        self._log_method("dmg.pool_create", kwargs)
+        data = self.dmg.pool_create(**kwargs)
+
+        if self.dmg.result.exit_status == 0:
+            # Convert the string of service replicas from the dmg command
+            # output into an ctype array for the DaosPool object using the
+            # same technique used in DaosPool.create().
+            service_replicas = [
+                int(value) for value in data["svc"].split(",")]
+            rank_t = ctypes.c_uint * len(service_replicas)
+            rank = rank_t(*list([svc for svc in service_replicas]))
```

Comment:
(pylint-consider-using-generator) Consider using a generator instead 'list(svc for svc in service_replicas)'
(pylint-unnecessary-comprehension) Unnecessary use of a comprehension, use list(service_replicas) instead.